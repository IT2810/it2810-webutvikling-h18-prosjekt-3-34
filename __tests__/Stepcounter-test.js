import React from "react";
import Stepcounter from ".././components/stepcounter";

import MockDate from "mockdate";

import renderer from "react-test-renderer";

//Snapshot test TODO: mock date
beforeAll(() => {
  MockDate.set("2018-10-10");
});

afterAll(() => {
  MockDate.reset();
});

test("renders correctly", () => {
  const tree = renderer.create(<Stepcounter viewDate={new Date()} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// Code gotten from expo docs. Only change is text shown. Not needed to test.
