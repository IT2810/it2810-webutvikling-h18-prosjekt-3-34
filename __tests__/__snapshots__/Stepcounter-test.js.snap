// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#eeeeee",
      "borderRadius": 4,
      "borderWidth": 1,
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 20,
      "maxHeight": 80,
      "padding": 20,
      "width": "85%",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
        "marginBottom": 12,
        "textAlign": "center",
      }
    }
  >
    Steps today: 
    0
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    StepGoal: 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    You have reached todays goal!
  </Text>
</View>
`;
